sql/plan/loop.go

```diff
@@ -18,25 +18,32 @@ import (
 	"fmt"
 	"io"
 	"strings"
+	"sync"
 
 	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
 )
 
 // Loop represents the LOOP statement, which loops over a set of statements.
 type Loop struct {
-	Label string
+	Label          string
+	Condition      sql.Expression // We continue looping until the condition returns false
```

Comment:
Seems like we need a `Resolved()` function in `Loop` now that will consider `Condition`, too.